luigi/task.py

```diff
@@ -575,6 +575,40 @@ def output(self):
         """
         return []  # default impl
 
+    def process_requires(self, requires):
+        """
+        Override in "template" tasks which themselves are supposed to be
+        subclassed and thus have their requires() overridden (name preserved to
+        provide consistent end-user experience), yet need to introduce
+        custom behaviour into dependencies.
+
+        Differences with :py:meth:`_requires`:
+          * method signature: :py:meth:`_requires` doesn't take params and calls
+            :py:meth:`requires` directly where as this method is expected to work only on
+            requirements passed via parameters.
+          * Code paths: :py:meth:`_requires` is run only during the statically defined
+            requirements where as this method works with dynamic dependencies yielded from
+            :py:meth:`run` also.
+
+        ex:
+            class ParentTask(luigi.Task):
+                foo = `bar`
+
+                def process_requires(self, requirements):
+                    for(req in flatten(requirements)):
+                        req.foo = self.foo
+
+                    return requirements
```

Comment:
what about just 1 line `return flatten(requirements) + self._paent_task_extra_requirements()`?

Reply:
This won't do. `process_requires` is for modifying the `requires` returned by the task, not add more requirements, which is already covered by the use-case described in `_requires`. Also using such an override with `process_requires` would result in same dependency being attempted multiple times in case of dynamic dependencies yielded from `run`.

Reply:
Well, you would also "modify" your dependency multiple times in case of dynamic dependencies right?

Can you aim to make this new function the "recommended replacement" for `_requires()`? It's probably just a very old mistake that we encourage users to override a private function like that.